src/main/kotlin/org/rust/ide/annotator/RsSyntaxErrorsAnnotator.kt

```diff
@@ -541,6 +542,16 @@ private fun checkWherePred(holder: AnnotationHolder, boundPred: RsWherePred) {
     }
 }
 
+private fun checkLambdaExpr(holder: AnnotationHolder, lambda: RsLambdaExpr) {
+    val asyncElement = lambda.node.findChildByType(RsElementTypes.ASYNC)
```

Comment:
Let's create an extension property `RsLambdaExpr.async` in `RsLambdaExpr.kt` near `val RsLambdaExpr.isAsync`